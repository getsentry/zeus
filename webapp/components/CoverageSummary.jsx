import React, {Component} from 'react';
import {Link} from 'react-router';
import PropTypes from 'prop-types';

import Collapsable from './Collapsable';
import {ResultGrid, Column, Header, Row} from './ResultGrid';

export default class CoverageSummary extends Component {
  static propTypes = {
    build: PropTypes.object.isRequired,
    coverage: PropTypes.arrayOf(PropTypes.object).isRequired,
    collapsable: PropTypes.bool,
    repo: PropTypes.object.isRequired,
    maxVisible: PropTypes.number
  };

  static defaultProps = {
    collapsable: false
  };

  constructor(props, context) {
    super(props, context);
    this.state = {collapsable: props.collapsable};
  }

  render() {
    let {build, coverage, repo} = this.props;
    let coverageLink = `/${repo.full_name}/builds/${build.number}/coverage`;

    return (
      <ResultGrid>
        <Header>
          <Column>File</Column>
          <Column width={80} textAlign="right">
            Diff
          </Column>
          <Column width={80} textAlign="right">
            Overall
          </Column>
        </Header>
        <Collapsable
          collapsable={this.props.collapsable}
          maxVisible={this.props.maxVisible}>
          {coverage.map(fileCoverage => {
            let totalDiffLines =
              fileCoverage.diff_lines_covered + fileCoverage.diff_lines_uncovered;
            let totalLines = fileCoverage.lines_covered + fileCoverage.lines_uncovered;

            return (
              <Row key={fileCoverage.filename}>
                <Column>
                  <Link
                    to={{pathname: coverageLink, query: {parent: fileCoverage.filename}}}>
                    {fileCoverage.filename}
                  </Link>
                </Column>
                <Column width={80} textAlign="right">
                  {totalDiffLines
                    ? `${parseInt(
                        (fileCoverage.diff_lines_covered / totalDiffLines) * 1000,
                        10
                      ) / 10}%`
                    : ''}
                </Column>
                <Column width={80} textAlign="right">
                  {totalLines
                    ? `${parseInt((fileCoverage.lines_covered / totalLines) * 1000, 10) /
                        10}%`
                    : ''}
                </Column>
              </Row>
            );
          })}
        </Collapsable>
      </ResultGrid>
    );
  }
}
