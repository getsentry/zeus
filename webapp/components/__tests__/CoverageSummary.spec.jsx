import React from 'react';
import {render} from 'enzyme';

import CoverageSummary from '../CoverageSummary';

const FIXTURE = [
  {
    data: 'NNNNNNNNNNNNNNNNNNCCNNNCCCCCUCCCNNCNNNUNNNNNNNNNNC',
    diff_lines_covered: 2.0,
    diff_lines_uncovered: 0.0,
    filename: 'webapp/components/Collapsable.jsx',
    lines_covered: 12.0,
    lines_uncovered: 2.0
  }
];

describe('CoverageSummary', () => {
  it('renders', () => {
    let context = TestStubs.standardContext();
    const tree = render(
      <CoverageSummary
        build={TestStubs.Build()}
        repo={TestStubs.Repository()}
        coverage={FIXTURE}
        location={{query: '', pathname: '/path-name'}}
      />,
      context
    );
    expect(tree).toMatchSnapshot();
  });
});
