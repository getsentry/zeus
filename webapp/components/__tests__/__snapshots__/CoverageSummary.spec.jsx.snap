// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoverageSummary renders 1`] = `
<div
  class="css-1mtyvxy-ResultGrid erefeu20"
>
  <div
    class="erefeu22 ResultGrid-Header css-1lhqp4g-Header"
  >
    <div
      class="erefeu21 ResultGrid-Column css-1oad8ub-Column"
    >
      File
    </div>
    <div
      class="erefeu21 ResultGrid-Column css-1vsx144-Column"
      style="text-align:right"
      width="80"
    >
      Diff
    </div>
    <div
      class="erefeu21 ResultGrid-Column css-1vsx144-Column"
      style="text-align:right"
      width="80"
    >
      Overall
    </div>
  </div>
  <div>
    <div
      class="erefeu23 ResultGrid-Row css-1d0bn69-Row"
    >
      <div
        class="erefeu21 ResultGrid-Column css-1oad8ub-Column"
      >
        <a>
          webapp/components/Collapsable.jsx
        </a>
      </div>
      <div
        class="erefeu21 ResultGrid-Column css-1vsx144-Column"
        style="text-align:right"
        width="80"
      >
        100%
      </div>
      <div
        class="erefeu21 ResultGrid-Column css-1vsx144-Column"
        style="text-align:right"
        width="80"
      >
        85.7%
      </div>
    </div>
  </div>
</div>
`;
